modules/delegates.js

```diff
@@ -247,23 +239,16 @@ __private.forge = function(cb) {
 	const lastBlock = modules.blocks.lastBlock.get();
 
 	if (currentSlot === slots.getSlotNumber(lastBlock.timestamp)) {
-		library.logger.debug('Waiting for next delegate slot');
+		library.logger.debug('Block already forged for the current slot');
 		return setImmediate(cb);
 	}
 
-	__private.getBlockSlotData(
+	__private.getDelegateKeypairForCurrentSlot(
 		currentSlot,
 		lastBlock.height + 1,
-		(err, currentBlockData) => {
-			if (err || currentBlockData === null) {
-				library.logger.warn('Skipping delegate slot', err);
-				return setImmediate(cb);
-			}
-
-			if (
-				slots.getSlotNumber(currentBlockData.time) !== slots.getSlotNumber()
-			) {
-				library.logger.debug('Delegate slot', slots.getSlotNumber());
+		(err, delegateKeypair) => {
+			if (err || delegateKeypair === null) {
+				library.logger.warn('Waiting for delegate slot', err);
```

Comment:
When there is `err` present here it comes from the SQL query, it should be logged separately with `error` level.

Reply:
Also `Waiting for delegate slot` should not be a `warn`, just `debug`, as this is completely normal behavior, nothing to worry about.